src/context/TokenContext.js

```diff
@@ -1,5 +1,54 @@
-import React from "react";
+import React, { createContext, useContext, useEffect, useState } from "react";
+import jwtDecode from "jwt-decode";
+
+const initialState = { session: null, user: null, username: null };
+export const AuthContext = createContext(initialState);
+
+const [token, setToken] = useState("");
+const [Authenticated, setAuthenticated] = useState(false);
```

Comment:
we can combine this into one state

`const [state, setState] = useState(initialState);`

Reply:
✅